storage/federated/ha_federated.cc

```diff
@@ -3044,6 +3059,18 @@ int ha_federated::real_query(const char *query, size_t length) {
     }
   }
 end:
+  /* Reconnection can take place above or inside cli_advanced_command().
+   If it happens in cli_advanced_command(), diagnositcs are may contain
+   the information about the error which triggered the reconnection.
+   But as the reconnection was fine, the information about transient
+   problems should not be propagated to the client.
+   */
+  if (!rc) {
+    Diagnostics_area *da = current_thd->get_stmt_da();
+    da->reset_condition_info(current_thd);
+    da->reset_diagnostics_area();
```

Comment:
Is it necessary to reset condition info here? Wouldn't just `current_thd->clear_error()` be sufficient? What do you say @dlenev ?